File: users/src/main/java/dental/core/users/modules/auth/services/AuthService.java
package dental.core.users.modules.auth.services;

import dental.core.users.modules.auth.dto.AuthResponse;
import dental.core.users.modules.auth.dto.ForgotPasswordRequest;
import dental.core.users.modules.auth.dto.LoginRequest;
import dental.core.users.modules.auth.dto.MessageResponse;
import dental.core.users.modules.auth.dto.RegisterRequest;
import dental.core.users.modules.auth.dto.ResetPasswordRequest;

/**
 * Servicio para manejar la autenticación de usuarios.
 * Proporciona métodos para login, registro y recuperación de contraseña.
 */
public interface AuthService {

    /**
     * Autentica un usuario con email y password.
     * 
     * @param loginRequest Datos de login (email y password)
     * @return AuthResponse con token JWT y datos del usuario
     */
    AuthResponse login(LoginRequest loginRequest);

    /**
     * Registra un nuevo usuario.
     * 
     * @param registerRequest Datos de registro (nombre, apellido, email, password, confirmPassword)
     * @return AuthResponse con token JWT y datos del usuario
     */
    AuthResponse register(RegisterRequest registerRequest);

    /**
     * Solicita recuperación de contraseña enviando un email con un token.
     * 
     * @param request Datos de la solicitud (email)
     * @return Mensaje de confirmación
     */
    MessageResponse forgotPassword(ForgotPasswordRequest request);

    /**
     * Resetea la contraseña usando el token recibido por email.
     * 
     * @param request Datos para resetear (token, nueva contraseña)
     * @return Mensaje de confirmación
     */
    MessageResponse resetPassword(ResetPasswordRequest request);
}

File: dental-portal/src/app/core/services/auth/auth.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AuthResponse, EmailAuthRequest, EmailRegisterRequest } from '../../../interfaces/auth/auth-response.interface';
import { BaseAuthService } from './base-auth.service';

export interface ForgotPasswordRequest {
  email: string;
}

export interface ResetPasswordRequest {
  token: string;
  newPassword: string;
  confirmPassword: string;
}

export interface MessageResponse {
  message: string;
}

@Injectable({
  providedIn: 'root',
})
export class AuthService extends BaseAuthService {
  private apiUrl: string = 'http://localhost:8081/api/users/auth';

  login(loginRequest: EmailAuthRequest): Observable<AuthResponse> {
    return this.handleAuthRequest(
      this.http.post<AuthResponse>(`${this.apiUrl}/login`, loginRequest)
    );
  }

  register(registerRequest: EmailRegisterRequest): Observable<AuthResponse> {
    return this.handleAuthRequest(
      this.http.post<AuthResponse>(`${this.apiUrl}/register`, registerRequest)
    );
  }

  forgotPassword(request: ForgotPasswordRequest): Observable<MessageResponse> {
    return this.http.post<MessageResponse>(`${this.apiUrl}/forgot-password`, request);
  }

  resetPassword(request: ResetPasswordRequest): Observable<MessageResponse> {
    return this.http.post<MessageResponse>(`${this.apiUrl}/reset-password`, request);
  }
}
